File: Practice/Practice/Leetcode/125_ValidPalindrome.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;

namespace Practice.Leetcode
{
    class _125_ValidPalindrome
    {
        public static void Main(String[] args)
        {
            string s = "A man, a plan, a canal: Panama";
            _125_ValidPalindrome a = new _125_ValidPalindrome();
            bool result = a.IsPalindrome(s);
        }
        public bool IsPalindrome(string s)
        {
            s = s.ToLower().Replace(" ", string.Empty);
            Regex rgx = new Regex("[^a-zA-Z0-9]");
            s = rgx.Replace(s,"");
            int start = 0;
            int end = s.Length - 1;
            while(end> start)
            {
                if (s[start] == s[end])
                {
                    start++;
                    end--;
                }
                else return false;
            }
            return true;
        }
    }
}

File: Practice/PythonApplication/LeetCode/Array/125_Valid_Palindrome.py
import re
def main():
    input = "A man, a plan, a canal: Panama"
    output = isPalindrome(input)
def isPalindrome(s):
    s = re.sub('[^A-Za-z0-9]', '', s)
    s = s.lower()
    start = 0
    end = len(s)-1
    while(end > start):
        if(s[start] == s[end]):
            start += 1
            end -=1
        else:
            return False
    return True

if __name__ == '__main__':
    main()


